src/plugins/release-monitor/index.js

```diff
@@ -0,0 +1,282 @@
+ * @param {Object} context - probot context object
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createSuccessPRStatus(context, sha) {
+    return createStatusOnPR({
+        context,
+        state: "success",
+        description: "No patch release is pending",
+        sha
+    });
+}
+
+/**
+ * Create pending status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createPendingPRStatus(context, sha) {
+    return createStatusOnPR({
+        context,
+        state: "pending",
+        description: "A patch release is pending",
+        sha
+    });
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context - probot context object
+ * @param {string} number - pr number
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function getAllCommitForPR(context, number) {
+    return context.github.pullRequests.getCommits(
+        context.repo({
+            number
+        })
+    );
+}
+
+/**
+ * Updates the status on all the provided prs
+ * @param {Object} context - probot context object
+ * @param {Array<Object>} prs - Collection of prs
+ * @param {boolean} isSuccess - to create a success status
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+async function createStatusOnPRs({ context, prs, isSuccess }) {
+    const statusFunc = isSuccess ? createSuccessPRStatus : createPendingPRStatus;
+
+    await Promise.all(prs.map(async pr => {
+        const allCommits = await getAllCommitForPR(context, pr.number);
+
+        await statusFunc(context, pluckLatestCommitSha(allCommits));
+    }));
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context - probot context object
+ * @param {boolean} isSuccess - to create a success status
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+async function createStatusOnAllPRs({ context, isSuccess }) {
+    const { data: allOpenPrs } = await getAllOpenPRs(context);
+
+    return createStatusOnPRs({
+        context,
+        prs: allOpenPrs,
+        isSuccess
+    });
+}
+
+/**
+ * Returns collection of all the PRs which do not have fix or docs tag on it
+ * @param {Object} context - probot context object
+ * @returns {Promise} promise
+ * @private
+ */
+async function getNonSemverPatchPRs(context) {
+    const { data: allOpenPrs } = await getAllOpenPRs(context);
+
+    return allOpenPrs.reduce(async(previousPromise, pr) => {
+        const coll = await previousPromise;
+        const allCommits = await getAllCommitForPR(context, pr.number);
+
+        if (!isMessageValidForPatchRelease(getCommitMessageForPR(allCommits, pr))) {
+            coll.push(pr);
+        }
+
+        return coll;
+    }, Promise.resolve([]));
```

Comment:
One alternative would be to use a loop here for readability:

```js
const nonSemverPatchPRs = [];
for (const pr of allOpenPrs) {
    const allCommits = await getAllCommitForPR(context, pr.number);
    if (!isMessageValidForPatchRelease(getCommitMessageForPR(allCommits, pr))) {
        nonSemverPatchPRs.push(pr);
    }
}

return nonSemverPatchPRs;
```

Alternatively, doing something like this would allow the requests to happen in parallel:

```js
return Promise
    .all(
        allOpenPrs.map(async pr => {
            const allCommits = await getAllCommitForPR(context, pr.number);
            return isMessageValidForPatchRelease(getCommitMessageForPR(allCommits, pr))
                ? []
                : [pr]
        }
    )
    .then(result =>
        result.reduce((accumulator, array) => accumulator.concat(array), [])
    );
```